Allow the draw function to be called programmatically

kwesterfeld opened the pull request:
This PR allows a user of eventdrops to call the draw() function programmatically, the way the zoom handler does.  We wish to alter the zoom behavior slightly and need a hook to call draw() from the requestAnimationFrame() in a similar way.

jpetitcolas commented:
Looks good. Can you rebase with your last PR to remove changes for JSPM?

Moreover, it miss a test to ensure we won't remove the `draw` method in the future. And, can you also update the docs? :)

kwesterfeld commented:
@jpetitcolas I just started looking at adding the test which would, as you say, help to keep the draw function from being removed.  I noticed recent commits have removed the .zoom() function, which we also depend on.  This was probably removed by mistake, ironically, because it is still documented in the README.

kwesterfeld commented:
Updated my branch to latest master, and resubmitted changes to add the hook for draw() and added back the hook for zoom().  I also added a test for these.

jpetitcolas commented:
It would need a last rebase. Yet, just wait a few ready PR to be merged, to ensure you won't have to rebase again. Will be embedded in the 0.3 version.

Sorry for the delay. :)

kwesterfeld commented:
No problem on the rebase, but how should I proceed?  Should I recreate a
new branch?  I think if I rebase and push --force the pull request will
auto close anyway.  Please advise.

On Mon, May 29, 2017 at 12:48 PM, Jonathan Petitcolas <
notifications@github.com> wrote:

> It would need a last rebase. Yet, just wait a few ready PR to be merged,
> to ensure you won't have to rebase again. Will be embedded in the 0.3
> version.
>
> Sorry for the delay. :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/marmelab/EventDrops/pull/119#issuecomment-304699541>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq8lxTS_w6xUishE2OwqHaEYIdmZs4Hks5r-vbKgaJpZM4L6_2->
> .
>

jpetitcolas commented:
@kwesterfeld: yes, just rebase and push force, it should not close the PR if it still contains some changes. :)

kwesterfeld commented:
@jpetitcolas I did a rebase and push force on this branch.  Can you review and include please?

Thanks

jpetitcolas commented on test/karma/eventDrops.js:
```diff
@@ -62,6 +62,16 @@ describe('eventDrops', () => {
         expect(div.querySelectorAll('.drop').length).toBe(3);
     });
 
+    it('should have a programmatic hooks', () => {
```
s/should have a programmatic hooks/should allow to call draw and zoom manually/

jpetitcolas commented on src/index.js:
```diff
@@ -57,11 +57,11 @@ function eventDrops(config = {}) {
 
             if (finalConfiguration.zoomable) {
-                zoom(svg, dimensions, scales, finalConfiguration);
+                this.zoom = zoom(svg, dimensions, scales, finalConfiguration, data);
```
Can you, if the chart is not zoomable, define `this.zoom` as an empty function `() => {}`? This way, it would always be a function, leading to an easier API.

fzaninotto commented:
No news for too long, closing
